File: backend-implicaction/src/main/java/com/dynonuggets/refonteimplicaction/model/RoleEnum.java
package com.dynonuggets.refonteimplicaction.model;

import com.dynonuggets.refonteimplicaction.exception.NotFoundException;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Arrays;

@Getter
@AllArgsConstructor
public enum RoleEnum {
    USER(1L, "ROLE_USER"),
    ADMIN(2L, "ROLE_ADMIN"),
    JOB_SEEKER(3L, "ROLE_JOB_SEEKER"),
    RECRUITER(4L, "ROLE_RECRUITER"),
    PREMIUM(5L, "ROLE_PREMIUM");

    private final Long id;
    private final String longName;

    public static RoleEnum byLongName(final String longName) {
        return Arrays.stream(RoleEnum.values())
                .filter(roleEnum -> roleEnum.getLongName().equals(longName))
                .findFirst()
                .orElseThrow(() -> new NotFoundException(String.format("role: %s not found.", longName)));
    }

}

File: frontend-implicaction/src/app/shared/enums/role.enum.ts
import {EnumCodeLabelAbstract} from './enum-code-label-abstract.enum';

export enum RoleEnumCode {
  ADMIN = 'ROLE_ADMIN',
  USER = 'ROLE_USER',
  PREMIUM = 'ROLE_PREMIUM'
}

export class RoleEnum extends EnumCodeLabelAbstract<RoleEnumCode> {
  static readonly ADMIN = new RoleEnum(RoleEnumCode.ADMIN, 'admin');
  static readonly USER = new RoleEnum(RoleEnumCode.USER, 'utilisateur');
  static readonly PREMIUM = new RoleEnum(RoleEnumCode.PREMIUM, 'premium');

  static all(): RoleEnum[] {
    return this.values();
  }

  static from(code: RoleEnumCode): RoleEnum {
    return this.fromCode(code);
  }
}
